File: ws/sisalema/u1/ws08CD_JavaClasses/ChickenFarmerSimulator/src-ec-edu-espe-farmsimulator/model/ChickenCoop.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.farmsimulator.model;

/**
 *
 * @author Yostin Sisalema,Code Masters, DCCO-ESPE
 */
public class ChickenCoop {
    private int id;
    private Chicken[] chickensCoop; 

    public ChickenCoop(int id, Chicken[] chickensCoop) {
        this.id = id;
        this.chickensCoop = chickensCoop;
    }

public String toString(String type) {
    StringBuilder chickensType = new StringBuilder();
    String chickenData = "";
    if (chickensCoop != null) {
        for (Chicken chicken : chickensCoop) {
            chickensType.append(chicken.toString(type)).append("\n");
        }
    }
    if (type.equals("csv")) { // csv
        chickenData = "ChickenCoop, id=" + id + ", chickensCoop=" + chickensType.toString();
    } else if (type.equals("txt")) { // txt
        chickenData = "ChickenCoop: id=" + id + ", chickensCoop=" + chickensType.toString()+"\n";
    } else if (type.equals("json")) { // json
        chickenData = "{\"ChickenCoop\": {\"id\": \"" + id + "\", \"chickensCoop\": \"" + chickensType.toString() + "\"}}"+"\n";
    }
    return chickenData;
}

        
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Chicken[] getChickens() {
        return chickensCoop;
    }

    public void setChickens(Chicken[] chickens) {
        this.chickensCoop = chickens;
    }
}

File: ws/sisalema/u1/ws08CD_JavaClasses/ChickenFarmSimulatorJavascript/farm_simulator/model/ChickenCoop.js
<<<<<<< HEAD
=======
<<<<<<< HEAD
>>>>>>> a064af6ecab12755c8c8b8a5858b342fd47aa80f
export class ChickenCoop {
    constructor(id, chickensCoop) {
        this.id = id;
        this.chickensCoop = chickensCoop;
    }

    toString(type) {
        let chickensType = "";
        if (this.chickensCoop != null) {
            for (let chicken of this.chickensCoop) {
                chickensType += chicken.toString(type) + "\n";
            }
        }
        
        let chickenData = "";
        if (type === "csv") { // csv
            chickenData = "ChickenCoop, id=" + this.id + ", \n" + chickensType;
        } else if (type === "txt") { // txt
            chickenData = "ChickenCoop: id=" + this.id + ", \n" + chickensType + "\n";
        } else if (type === "json") { // json
            chickenData = "{\"ChickenCoop\": {\"id\": \"" + this.id + "\", \"\n" + chickensType + "\"}}\n";
        }
        return chickenData;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getChickens() {
        return this.chickensCoop;
    }

    setChickens(chickens) {
        this.chickensCoop = chickens;
    }
}
<<<<<<< HEAD
=======
=======
export class ChickenCoop {
    constructor(id, chickensCoop) {
        this.id = id;
        this.chickensCoop = chickensCoop;
    }

    toString(type) {
        let chickensType = "";
        if (this.chickensCoop != null) {
            for (let chicken of this.chickensCoop) {
                chickensType += chicken.toString(type) + "\n";
            }
        }
        
        let chickenData = "";
        if (type === "csv") { // csv
            chickenData = "ChickenCoop, id=" + this.id + ", \n" + chickensType;
        } else if (type === "txt") { // txt
            chickenData = "ChickenCoop: id=" + this.id + ", \n" + chickensType + "\n";
        } else if (type === "json") { // json
            chickenData = "{\"ChickenCoop\": {\"id\": \"" + this.id + "\", \"\n" + chickensType + "\"}}\n";
        }
        return chickenData;
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getChickens() {
        return this.chickensCoop;
    }

    setChickens(chickens) {
        this.chickensCoop = chickens;
    }
}
>>>>>>> 8e40484f0077d7fe10786aa1aa0f3cc51debe068
>>>>>>> a064af6ecab12755c8c8b8a5858b342fd47aa80f
